File: example-java/src/main/java/dev/vkrasnousov/restring/example/SampleApplication.java
package dev.vkrasnousov.restring.example;

import android.app.Application;

import java.util.Arrays;
import java.util.List;
import java.util.Locale;

import dev.b3nedikt.app_locale.AppLocale;
import dev.b3nedikt.reword.RewordInterceptor;
import dev.b3nedikt.viewpump.ViewPump;
import dev.vkrasnousov.restring.Restring;


public class SampleApplication extends Application {

    @Override
    public void onCreate() {
        super.onCreate();

        AppLocale.setSupportedLocales(APP_LOCALES);

        Restring.init(this);
        ViewPump.init(RewordInterceptor.INSTANCE);
    }

    private static final List<Locale> APP_LOCALES =
            Arrays.asList(Locale.ENGLISH, Locale.US, Locales.LOCALE_AUSTRIAN_GERMAN);
}

File: example/src/main/java/dev/vkrasnousov/restring/example/SampleApplication.kt
package dev.vkrasnousov.restring.example

import android.app.Application
import dev.b3nedikt.app_locale.AppLocale
import dev.b3nedikt.reword.RewordInterceptor
import dev.b3nedikt.viewpump.ViewPump
import dev.vkrasnousov.restring.Restring
import dev.vkrasnousov.restring.example.Locales.LOCALE_AUSTRIAN_GERMAN
import java.util.*


class SampleApplication : Application() {

    override fun onCreate() {
        super.onCreate()

        AppLocale.supportedLocales = listOf(Locale.ENGLISH, Locale.US, LOCALE_AUSTRIAN_GERMAN)

        Restring.init(this)
        Restring.localeProvider = AppLocaleLocaleProvider

        ViewPump.init(RewordInterceptor)
    }
}